opennms-dao-api/src/main/java/org/opennms/netmgt/dao/api/OnmsDao.java

```diff
@@ -57,6 +57,13 @@
 
     void delete(K key);
 
+    /**
+     * Deletes all entities of this type - use with care!
+     */
+    default void deleteAll(){
```

Comment:
Default implementations don't play nice with OSGi, so we may run into some problems here.